app/services/degrees/dfe_reference.rb

```diff
@@ -44,14 +82,20 @@ def find_item(list_type, filters)
 
     private
 
+      def match_degree_strings(filters, item)
+        filters.compact.any? do |_, value|
+          match_values(item).include?(value.downcase.strip)
+        end
+      end
+
       def match_values(item)
         [item[:name], item[:match_synonyms], item[:abbreviation]].flatten.compact.map(&:downcase).map(&:strip)
       end
 
-      def match_degree_strings(filters, item)
-        filters.compact.any? do |_field, value|
-          match_values(item).include?(value.downcase.strip)
-        end
+      # The HESA code can sometimes be padded with some leading zeros.
+      # So we need to sensibly convert the strings to their integer values for an equal comparison.
+      def sanitised_hesa_code(code)
```

Comment:
Maybe instead of using the comment we could rename the method? I think something as simple as `remove_leading_zeros` could work or `remove_leading_zeros_from_hesa_code` if we want to be specific? What do you think?